geode-apis-compatible-with-redis/src/commonTest/java/org/apache/geode/redis/GeodeRedisServerRule.java

```diff
@@ -49,6 +49,12 @@ protected void before() {
     cache = cacheFactory.create();
     server = new GeodeRedisServer("localhost", 0, (InternalCache) cache);
     server.setAllowUnsupportedCommands(enableUnsupportedCommands);
+
+    // Ensure that buckets are created up front
+    try {
```

Comment:
I think this block should be refactored into a method on GeodeRedisServer.
Also it looks to me like it will be happy even if a bucket can not be created. It looks like we basically need to call SlotAdvisor.getMemberInfo for each bucketId but that method has some retry logic in it and it can fail to get the member info. In that case it logs an error and returns null but this code ends up ignoring those nulls and moving on. I don't know if you want something different here that would fail the startup if one of the memberInfo calls times out.

Reply:
Regarding the error case where the member info cannot be retrieved - that would result in a 'non-existent' slot (or range of slots) at which point any clients should produce some kind of error if they were trying to write to a missing slot.

Regarding pulling this into `GeodeRedisServer` - are you thinking it should run as part of startup? If so, I'd be wary of that since it would create a bucket imbalance unless we also did a rebalance as part of that lifecycle. As it is, any cluster-aware client will always retrieve the topology as the first thing it does which would implicitly create the buckets. It's still a problem if all members are not up before the very first client starts to connect.